File: backend/src/main/kotlin/com/example/splitandpay/backend/model/dto/RoomDto.kt
package com.example.splitandpay.backend.model.dto

import com.example.splitandpay.backend.serialization.LocalDateTimeSerializer
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class RoomDto(
    val id: Long,
    val name: String,
    val owner: OwnerDto,
    @Serializable(with = LocalDateTimeSerializer::class)
    val createdAt: LocalDateTime,
    val users: List<OwnerDto>? = null,
    val receipt: List<ProductDto>? = null,
    val totalSum: Double? = null,
)

File: react/src/model/RoomDto.ts
import OwnerDto from "./OwnerDto";
import ProductDto from "./ProductDto";

export default interface RoomDto {
    id: number,
    name: string,
    owner: OwnerDto,
    createdAt: string,
    users: OwnerDto[] | null,
    receipt: ProductDto[] | null,
    totalSum: number | null
}